### ISSUE TITLE
Fix S3358 FP: Suggesting to extract ternary from EF Core select

### ISSUE THREAD

**Issue by grosch-intl** [SonarSource/sonar-dotnet#9801]
### Description

The below code is not able to have the ternary operator extracted as it's part of a linq query, so the whole thing has to be run against each record.

### Reproducer

```cs
        var query = context.Checklist
            .AsNoTracking()
            .Where(...)
            .Select(x => new {
                ReviewStatus = x.ReviewApprovedUtc.HasValue ? "Approved" : x.ReviewRejectedUtc.HasValue ? "Rejected" : x.ReviewRequestedUtc.HasValue ? "Pending" : null,              
```

### Product and Version

SonarAnalyzer.CSharp 10.23.0.137933

**Comment by martin-strecker-sonarsource** [comment]
Hi @grosch-intl, thanks for the report — confirmed as a false positive. Inside an `IQueryable<T>.Select(...)` lambda, the C# compiler converts the body to an `Expression<Func<T, TResult>>`, where statement-bodied lambdas aren't legal (CS0834) — so the rule's "extract into an independent statement" advice can't be followed.

Tracked internally as NET-3878 and a fix is up for review. It will roll out with the next analyzer release.

**Comment by martin-strecker-sonarsource** [comment]
The issue was fixed in 7596b500fe0f7ba2ebca1ef592b3628bb5024889 and will be part of the next release. Thank you @grosch-intl for the report!
